azure/defaults.go

```diff
@@ -283,13 +283,21 @@ func AvailabilitySetID(subscriptionID, resourceGroup, availabilitySetName string
 }
 
 // GetDefaultImageSKUID gets the SKU ID of the image to use for the provided version of Kubernetes.
-// note: osAndVersion is expected to be in the format of {os}-{version} (ex: unbuntu-2004 or windows-2022)
+// note: osAndVersion is expected to be in the format of {os}-{version} (ex: ubuntu-2004 or windows-2022)
 func getDefaultImageSKUID(k8sVersion, osAndVersion string) (string, error) {
-	version, err := semver.ParseTolerant(k8sVersion)
+	v, err := semver.ParseTolerant(k8sVersion)
 	if err != nil {
 		return "", errors.Wrapf(err, "unable to parse Kubernetes version \"%s\" in spec, expected valid SemVer string", k8sVersion)
 	}
-	return fmt.Sprintf("k8s-%ddot%ddot%d-%s", version.Major, version.Minor, version.Patch, osAndVersion), nil
+	// Old SKU versions before 1.21.12, 1.22.9, or 1.23.6 are named like "k8s-1dot21dot2-ubuntu-2004".
+	if (v.Major == 1 && v.Minor < 21) ||
```

Comment:
Not sure if this is still part of the WIP but we can't use `LatestVersion` for the image version anymore since the version is per k8s version now, we're going to have set a specific version. 

Which also makes me realize we're going to have to be fetching images with the vm image API regardless, so might as well do discovery instead of hardcoding a set of versions for the new naming scheme, wdyt?